### PR TITLE
Fix/copy version create date

### PR DESCRIPTION
**Pull request by Davidmattei**
|Q              |A  |
|---------------|---|
|Bug fix?       |y|
|New feature?   |y|	
|BC breaks?     |n|	
|Deprecations?  |n|	
|Fixed tickets  |n|	

Add clear_on_copy extra option for computed fieldTypes

Requires: 

https://github.com/ems-project/EMSCommonBundle/pull/218

### CODE REVIEW

File: src/Form/DataField/ComputedFieldType.php [hunk 1]
```diff
@@ -94,6 +94,11 @@ public function buildOptionsForm(FormBuilderInterface $builder, array $options)
         ->add('copy_to', TextType::class, [
                 'required' => false,
         ]);
+
+        $optionsForm->get('extraOptions')->add('clear_on_copy', CheckboxType::class, [
+            'required' => false,
+        ]);
```
Comment (Davidmattei) [inline src/Form/DataField/ComputedFieldType.php:100]: Move to DataFieldType

File: src/Entity/ContentType.php [hunk 2]
```diff
@@ -1057,6 +1057,25 @@ public function getFieldType()
...
+    /**
+     * @return string[]
+     */
+    public function getClearOnCopyProperties(): array
+    {
+        $clearPropertyNames = [];
+
+        foreach ($this->getFieldType()->loopChildren() as $child) {
+            $extraOptions = $child->getExtraOptions();
+            $clearOnCopy = isset($extraOptions['clear_on_copy']) ? \boolval($extraOptions['clear_on_copy']) : false;
```
Comment (theus77) [inline src/Entity/ContentType.php:1069]: Shorter solution? If $extraOptions['clear_on_copy'] is a boolean and not a string 'true'
```suggestion

            $clearOnCopy = true === $extraOptions['clear_on_copy'] ?? null;
```

File: src/Entity/FieldType.php [hunk 3]
```diff
@@ -665,6 +665,17 @@ public function getChildren()
...
 
+    /**
+     * @return \Generator|FieldType[]
+     */
+    public function loopChildren(): \Generator
+    {
+        foreach ($this->children as $child) {
+            yield $child;
+            yield from $child->loopChildren();
+        }
```
Comment (theus77) [inline src/Entity/FieldType.php:676]: Nice depth first walk solution

**Review by theus77 — APPROVED** [review]

**Review by theus77 — APPROVED** [review]

File: src/Entity/ContentType.php (hunk 2 above)
Comment (Davidmattei) [reply src/Entity/ContentType.php]: Agree

File: src/Entity/FieldType.php (hunk 3 above)
Comment (Davidmattei) [reply src/Entity/FieldType.php:676]: Yes very powerful !

**Pull request merged**